### PR TITLE
Reorganize functions into api and types modules

### PR DESCRIPTION
**Pull request by jreese**

### CODE REVIEW

File: usort/__init__.py [hunk 1]
```diff
@@ -3,7 +3,19 @@
...
+
 try:
-    from .version import version as __version__  # noqa: F401
+    from .version import version as __version__
 except ImportError:
     __version__ = "dev"
+
+from .api import usort_bytes, usort_path, usort_stdin, usort_string
+
+# DEPRECATED: preserve old api, will be removed by 1.0
```
Comment (thatch) [inline usort/__init__.py:15]: It should be safe to have `usort.sorting` do the imports from `.api`, why do you need to do assignment here? This is a little weird.

File: usort/api.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+# LICENSE file in the root directory of this source tree.
+
+import sys
+from pathlib import Path
+from typing import Iterable, Optional, Tuple
+
+from .config import Config
+from .sorting import ImportSortingTransformer
+from .types import Result
+from .util import try_parse, timed, walk
```
Comment (thatch) [inline usort/api.py:13]: It's probably worth defining an `__all__` to avoid things like `Path` or `timed` sneaking in if someone does a star-import

**Review by thatch — APPROVED** [review]
Moving things around is fine, and I see tests still pass.  A couple of minor comments.

File: usort/__init__.py (hunk 1 above)
Comment (jreese) [reply usort/__init__.py:15]: That causes a circular import because usort_string needs something from sorting.

Comment (thatch) [reply usort/__init__.py:15]: Then I'd prefer usort/sorting.py contain wrappers so the temporary existence, and future deletion, is more obvious:

```py
# DEPRECATED: preserve old api, will be removed by 1.0
def usort_bytes(...):
  from .api import usort_bytes
  return usort_bytes(...)
```

**Pull request merged**